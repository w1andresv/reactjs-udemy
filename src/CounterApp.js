
import PropTypes from 'prop-types';
import React, { useState } from 'react';

const CounterApp = ({ value=10 }) => {

   const [counter, setCounter] = useState(value);


   const clickAdd = (e) => {
       setCounter(counter+1)
   }
   const clickSubstrac = (e) => {
       setCounter(counter-1)
   }
   const clickReset = (e) => {
       setCounter(value)
   }
   return (
      <>
         <h1>Contador</h1>
         <h2>{counter}</h2>
         <button id='add' onClick={clickAdd}>+1</button>
         <button id='reset' onClick={clickReset}>Reset</button>
         <button id='subs' onClick={clickSubstrac}>-1</button>
         <input></input>
      </>
   )
}

CounterApp.propTypes = {
   value: PropTypes.number
}

export default CounterApp