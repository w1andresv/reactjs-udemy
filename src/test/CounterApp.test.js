import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import React from 'react';
import CounterApp from "../CounterApp";

describe('Pruebas en <CounterApp />', () => {
   let wrapper = shallow(<CounterApp />);

   beforeEach(() => {
      wrapper = shallow(<CounterApp />);
   });
   test('Renderizar componente', () => {
      expect(wrapper).toMatchSnapshot();
   });

   test('Renderizar componente con valor de 123', () => {
      const valor = 123;
      const wrapper = shallow(<CounterApp value={valor} />);
      const counter = wrapper.find('h2').text();
      expect(counter).toBe('123');
   });


   test('should boton add', () => {
      wrapper.find('#add').simulate('click');
      const counter = wrapper.find('h2').text();
      expect(counter).toBe('11');
   })
   test('should boton subs', () => {
      wrapper.find('#subs').simulate('click');
      const counter = wrapper.find('h2').text();
      expect(counter).toBe('9');
   })
   test('should boton reset', () => {
      const valor = 123;
      const wrapper = shallow(<CounterApp value={valor} />);
      wrapper.find('#add').simulate('click');
      wrapper.find('#reset').simulate('click');
      const counter = wrapper.find('h2').text();
      expect(counter).toBe('123');
   })
});