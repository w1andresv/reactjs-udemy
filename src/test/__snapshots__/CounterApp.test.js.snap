// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> Renderizar componente 1`] = `
<Fragment>
  <h1>
    Contador
  </h1>
  <h2>
    10
  </h2>
  <button
    id="add"
    onClick={[Function]}
  >
    +1
  </button>
  <button
    id="reset"
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    id="subs"
    onClick={[Function]}
  >
    -1
  </button>
  <input />
</Fragment>
`;
